packages/client/hmi-client/src/components/workflow/ops/calibrate-ciemss/tera-calibrate-node-ciemss.vue

```diff
@@ -150,118 +142,26 @@ async function updateLossChartWithSimulation() {
-	});
-	state.mapping.forEach((mapObj) => {
-		reverseMap[mapObj.datasetVariable] = 'Observations';
-	});
-
-	// Need to get the dataset's time field
-	const datasetTimeField = state.timestampColName;
-	const dateOptions = getVegaDateOptions(model.value, modelConfiguration.value);
-
-	const variableCharts = selectedVariableSettings.value.map((setting) => {
-		const variable = setting.selectedVariables[0];
-		const datasetVariables: string[] = [];
-		const mapObj = state.mapping.find((d) => d.modelVariable === variable);
-		if (mapObj) {
-			datasetVariables.push(mapObj.datasetVariable);
-		}
-		const annotations = getChartAnnotationsByChartId(setting.id);
-
-		// variable chart
-		const chart = createForecastChart(
-			{
-				data: result,
-				variables: [`${pyciemssMap[variable]}:pre`, pyciemssMap[variable]],
-				timeField: 'timepoint_id',
-				groupField: 'sample_id'
-			},
-			{
-				data: resultSummary,
-				variables: [`${pyciemssMap[variable]}_mean:pre`, `${pyciemssMap[variable]}_mean`],
-				timeField: 'timepoint_id'
-			},
-			{
-				data: groundTruth.value,
-				variables: datasetVariables,
-				timeField: datasetTimeField as string
-			},
-			{
-				title: '',
-				legend: true,
-				translationMap: reverseMap,
-				xAxisTitle: modelVarUnits.value._time || 'Time',
-				yAxisTitle: modelVarUnits.value[variable] || '',
-				colorscheme: ['#AAB3C6', '#1B8073'],
-				...chartSize,
-				dateOptions
-			}
-		);
-		applyForecastChartAnnotations(chart, annotations);
-		chart.layer.push(...createInterventionChartMarkers(groupedInterventionOutputs.value[variable]));
-
-		return chart;
-	});
-
-	// intervention charts
-	const interventionCharts = Object.keys(groupedInterventionOutputs.value).map((key) => {
-		const chart = createForecastChart(
-			{
-				data: result,
-				variables: [pyciemssMap[key]],
-				timeField: 'timepoint_id',
-				groupField: 'sample_id'
-			},
-			null,
-			{
-				data: groundTruth.value,
-				variables: [key],
-				timeField: datasetTimeField as string
-			},
-			{
-				title: key,
-				legend: true,
-				translationMap: reverseMap,
-				xAxisTitle: modelVarUnits.value._time || 'Time',
-				yAxisTitle: modelVarUnits.value[key] || '',
-				colorscheme: ['#AAB3C6', '#1B8073'],
-				...chartSize,
-				dateOptions
-			}
-		);
-		chart.layer.push(...createInterventionChartMarkers(groupedInterventionOutputs.value[key]));
-		return chart;
-	});
-
-	return { variableCharts, interventionCharts };
-});
-
-const { getChartAnnotationsByChartId } = useChartAnnotations(props.node.id);
+const { selectedVariableSettings, selectedInterventionSettings } = useChartSettings(props, emit);
+const { useInterventionCharts, useVariableCharts } = useCharts(
+	props.node.id,
+	model,
+	modelConfiguration,
+	preparedChartInputs,
+	toRef(chartSize),
+	computed(() => interventionPolicy.value?.interventions ?? [])
+);
+const toDatasetVar = (modelVar: string) => mapModelVarToDatasetVar(props.node.state, modelVar);
```

Comment:
Same as my previous comment

If this is only being used for output I suspect we want to use the output's state's mapping here

Reply:
Done